src/corelibs/U2Formats/src/TextDocumentFormat.cpp

```diff
@@ -48,17 +49,70 @@ FormatCheckResult TextDocumentFormat::checkRawData(const QByteArray &rawData, co
     return checkResult;
 }
 
-Document *TextDocumentFormat::loadDocument(IOAdapter *io, const U2DbiRef &dbiRef, const QVariantMap &fs, U2OpStatus &os) {
+Document *TextDocumentFormatDeprecated::loadDocument(IOAdapter *io, const U2DbiRef &dbiRef, const QVariantMap &fs, U2OpStatus &os) {
     io->setFormatMode(IOAdapter::TextMode);
     Document *doc = loadTextDocument(io, dbiRef, fs, os);
 
     return doc;
 }
 
-DNASequence *TextDocumentFormat::loadTextSequence(IOAdapter *io, U2OpStatus &ti) {
+DNASequence *TextDocumentFormatDeprecated::loadTextSequence(IOAdapter *io, U2OpStatus &ti) {
     Q_UNUSED(io);
     ti.setError("This document format does not support streaming reading mode");
     return NULL;
 }
 
+///////////////////////////////////////////
+/// TextDocumentFormat
+///////////////////////////////////////////
+
+TextDocumentFormat::TextDocumentFormat(QObject *p, const DocumentFormatId &id, DocumentFormatFlags _flags, const QStringList &fileExts)
+    : DocumentFormat(p, id, _flags, fileExts) {
+}
+
+FormatCheckResult TextDocumentFormat::checkRawData(const QByteArray &rawBinaryData, const GUrl &url) const {
+    QTextStream stream(rawBinaryData, QIODevice::ReadOnly);    // Use QTextStream to auto-detect multi-byte encoding.
+    QString text = stream.readAll();
```

Comment:
Byte Order Marks is still a problem, I suppose

Reply:
I support not.

Reply:
You suppose or you have checked it?

Reply:
Hope this will help: https://doc.qt.io/qt-5/qtextstream.html.